plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1438,6 +1438,78 @@ producer_plugin::get_account_ram_corrections( const get_account_ram_corrections_
    return result;
 }
 
+producer_plugin::get_unapplied_transactions_result
+producer_plugin::get_unapplied_transactions( const get_unapplied_transactions_params& p, const fc::time_point& deadline ) const {
+
+   fc::microseconds params_time_limit = p.time_limit_ms ? fc::milliseconds(*p.time_limit_ms) : fc::milliseconds(10);
+   fc::time_point params_deadline = fc::time_point::now() + params_time_limit;
+
+   auto& ua = my->_unapplied_transactions;
+
+   auto itr = ([&](){
+      if (!p.lower_bound.empty()) {
+         try {
+            auto trx_id = transaction_id_type( p.lower_bound );
+            return ua.lower_bound( trx_id );
+         } catch( ... ) {
+            return ua.end();
+         }
+      } else {
+         return ua.begin();
+      }
+   })();
+
+   auto get_trx_type = [&](trx_enum_type t, bool read_only) {
+      if( read_only ) return "read_only";
+      switch( t ) {
+         case trx_enum_type::unknown:
+            return "unknown";
+         case trx_enum_type::persisted:
+            return "persisted";
+         case trx_enum_type::forked:
+            return "forked";
+         case trx_enum_type::aborted:
+            return "aborted";
+         case trx_enum_type::incoming:
+            return "incoming";
+      }
+      return "unknown type";
+   };
+
+   get_unapplied_transactions_result result;
+   result.size = ua.size();
+   result.incoming_size = ua.incoming_size();
+
+   uint32_t remaining = p.limit ? *p.limit : std::numeric_limits<uint32_t>::max();
```

Comment:
Defaulting to max<unit32> seems big. Not sure what is typical, I would have made the default a max of short int, uint16. Max should be bigger up to unit32.

Reply:
It is limited by the `time_limit_ms`, setting `limit` to max uint32 just means it is limited by time only.